server/conn.go

```diff
@@ -1762,16 +1778,36 @@ func (cc *clientConn) handleChangeUser(ctx context.Context, data []byte) error {
 	if err != nil {
 		return err
 	}
+	return cc.handleCommonConnectionReset()
+}
+
+func (cc *clientConn) handleResetConnection(ctx context.Context) error {
+	err := cc.ctx.Close()
+	if err != nil {
+		logutil.Logger(ctx).Debug("close old context failed", zap.Error(err))
+	}
+	var tlsStatePtr *tls.ConnectionState
+	if cc.tlsConn != nil {
+		tlsState := cc.tlsConn.ConnectionState()
+		tlsStatePtr = &tlsState
+	}
+	cc.ctx, err = cc.server.driver.OpenCtx(uint64(cc.connectionID), cc.capability, cc.collation, cc.dbname, tlsStatePtr)
+	if err != nil {
+		return err
+	}
```

Comment:
maybe we need to assign more info for the new session(e.g. `sessionVars.{User,ActiveRoles}`, `GetSessionManager` ), just like `openSessionAndDoAuth`?

Reply:
Thanks, I'll take a look. I just modified my example program and it looks like the privileges and currently selected database are lost :(

Reply:
This has been addressed. I added a session method `AuthWithoutVerification`.